modules/custom/dumprunparams/main.nf

```diff
@@ -0,0 +1,45 @@
+def VERSION="0.0.1"
+
+process CUSTOM_DUMPRUNPARAMS {
+    label 'process_low'
+
+    input:
+    val(exclude)
+
+    output:
+    path "params_mqc.tsv", emit: mqc_tsv
+    path "versions.yml"  , emit: versions
+
+    script:
+    run_params = params
+    to_exclude = exclude instanceof List && exclude.size() > 0 ? exclude.join('|') : ''
+    exclude_cmd = exclude ? "| sed -E '/${to_exclude}/d'" : ''
+    println exclude
+    println exclude_cmd
+    """
+    echo "${run_params}" > run_params.txt
+    table_header=\$( sed 's/,/\t/g' <( echo "Parameter,col1" ) )
+
+    ## Remove outer brackets, remove nested arrays, split each param new line, delete lines containing params to ignore, convert params to cli flags
+    run_params=\$( sed 's#^\\[##g;s#\\]\$##g' run_params.txt | sed 's/\\[.*]/\\.\\.\\.truncated_list\\.\\.\\./g' | sed 's#, #\\n#g;s#:#\t#g' ${exclude_cmd} | sed 's#^#--#g' )
```

Comment:
:exploding_head: 

I guess this works, but this looks like a nightmare to debug if anything goes wrong. 
Since `params` is already a groovy map, couldn't you just dump it to yaml directly in groovy using an `exec` section? 

Take a look at: https://github.com/nf-core/modules/blob/edbbbbf42097c0696305dd2fd6d0e1070a7dbb07/modules/jupyternotebook/parametrize.nf#L22-L32

Reply:
Definitely up for removing the whole `sed` mess! Will try it out!

Reply:
Ok, getting halfway there but it _will_ need some form of `sed` mess to format it in the way MQC wants it... but hopefully less :+1:

Reply:
You have it as a structured variable in Groovy here, so my inclination would be to keep the entire process as pure groovy without a `script` block. Make a deep copy, restructure as needed and then write out a YAML / JSON file. MultiQC can handle these just as well as a tsv / csv.

Reply:
eg. Flatten the nested params: https://stackoverflow.com/a/51288381 and stick in a new object under a `data` key. 

Then add the MultiQC config keys and you're away..

Reply:
I was leaning towards that so :+1: 

the only problem is the nested params will be extremely messy to display - for example in the tests (and I suspect also igenomes) the reference genomes wiht all the sub directories would be completely incomprhensiable. But maybe that's OK given I'm trying to have the `exclude` functionality.

Reply:
Yeah exactly - we exclude those when rendering `--help` for the same reason. Might even exclude anything nested, not sure now 🤔